persist/fs/commitlog/iterator.go

```diff
@@ -185,6 +201,22 @@ func (i *iterator) nextReader() bool {
 	return true
 }
 
+func filterFiles(files []string, pred ReadCommitLogPredicate) ([]string, error) {
+	filteredFiles := []string{}
+	var multiErr xerrors.MultiError
+	for _, file := range files {
+		fileTime, err := fs.TimeFromFileName(file)
```

Comment:
It might be worth actually reading the info struct instead of relying on file name?  File name is usually only for ordering, not for relying on data in the actual file itself.

Reply:
Hey Rob, I did this refactoring. Couple of notes:

1) I wasn't able to share too much of the logic between the reader itself and the new ReadLogInfo function. Thats primarily because the Reader methods are highly entangled with its state (pooling, shared buffers, etc etc). I tried doing it and it just made the code unnecessarily hard to grok. I think its ok though because the new ReadLogInfo function is fairly short, straightforward, and doesn't have too much business logic in int. Its just setting up a chunk reader, opening/closing a file, instantiating a msgpack decoder etc etc.

2) Previously we were relying on the Options struct to tell us how long of a period of time the commitlog represented (via the Blocksize). I think this would have caused trouble for people who tried to change this value and restart a node, so now that we get that info from the commitlog itself I think the logic is more sound.

3) I was unable to move the call to ReadLogInfo into the fs.go file because there is already a dependent in the commitlog reader on the fs package so it would have become cyclic. With that said, it seems reasonable for the FS package to have functions for determining information about the commitlog from the filename, and for the commitlog package to have functions for determining information about the commitlog by parsing it.

4) Presumably we wanted to make this change because we trust the loginfo struct more than we trust the filename? Should I look at both? We could make the predicate pass both values so that if there is a discrepancy you could choose to read the file anyways, but that might be overkill.